payment/src/main/java/org/killbill/billing/payment/core/sm/control/OperationControlCallback.java

```diff
@@ -99,16 +100,14 @@ public OperationResult doOperationCallback() throws OperationException {
                                                                                                      paymentStateControlContext.isApiPayment(),
                                                                                                      paymentStateContext.getCallContext());
 
-                final PriorPaymentControlResult pluginResult;
                 try {
-                    pluginResult = executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
-                    if (pluginResult != null && pluginResult.isAborted()) {
-                        // Transition to ABORTED
-                        return PluginDispatcher.createPluginDispatcherReturnType(OperationResult.EXCEPTION);
-                    }
+                    executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
                 } catch (final PaymentControlApiException e) {
                     // Transition to ABORTED and throw PaymentControlApiException to caller.
                     throw new OperationException(e, OperationResult.EXCEPTION);
+                } catch (final PaymentApiException paymentAbortedException) {
+                    // Transition to ABORTED
+                    throw new OperationException(paymentAbortedException, OperationResult.EXCEPTION);
```

Comment:
I think that works, but i would have preferred to not change the signature of `executePluginPriorCalls` (which now throws a PaymentApiException solely to support that case), and instead just throw the OperationException (with inside the PaymentApiException(ErrorCode. PAYMENT_PLUGIN_API_ABORTED) in the few lines above.

However, i think i understand why you did it that way: Probably you wanted to keep track of which plugin aborted (see remark on the message above).